File: yht0827/1월/1월 24일/행렬 테두리 회전하기.java
class Solution {
	int[][] map;

	public int[] solution(int rows, int columns, int[][] queries) {
		int[] answer = new int[queries.length]; // 정답 개수 배열
		this.map = new int[rows][columns]; // 맵 생성

		// map 초기화
		for (int i = 0; i < rows; i++) {
			for (int j = 0; j < columns; j++) {
				map[i][j] = i * columns + j + 1;
			}
		}

		// 회전 후 최소값 저장
		for (int i = 0; i < queries.length; i++) {
			answer[i] = move(queries[i]);
		}

		return answer;
	}

	// 회전 순서 위 왼쪽 아래 오른쪽 역순으로 진행
	public int move(int[] info) {

		int x1 = info[0] - 1; // 시작 범위 x
		int y1 = info[1] - 1; // 시작 범위 y
		int x2 = info[2] - 1; // 끝 범위 x
		int y2 = info[3] - 1; // 끝 범위 y

		int start = this.map[x1][y1]; //시작 위치 값 임시 저장
		int min = start; // 최소 값 저장 처음에는 시작 위치 값으로 저장

		// 아래에서 위
		for (int i = x1; i < x2; i++) {
			this.map[i][y1] = this.map[i + 1][y1];
			// 최소 값 비교
			if (min > this.map[i][y1]) {
				min = this.map[i][y1];
			}
		}

		// 오른쪽에서 왼쪽
		for (int i = y1; i < y2; i++) {
			this.map[x2][i] = this.map[x2][i + 1];
			if (min > this.map[x2][i]) {
				min = this.map[x2][i];
			}
		}

		// 위에서 아래
		for (int i = x2; i > x1; i--) {
			this.map[i][y2] = this.map[i - 1][y2];
			if (min > this.map[i][y2])
				min = this.map[i][y2];
		}
		// 왼쪽에서 오른쪽
		for (int i = y2; i > y1; i--) {
			this.map[x1][i] = this.map[x1][i - 1];
			if (min > this.map[x1][i])
				min = this.map[x1][i];
		}

		this.map[x1][y1 + 1] = start; // 처음 위치 저장

		return min; // 최소값 반환
	}
}

File: zuzubibi/1월/1월 24일/행렬 테두리 회전하기.py
from collections import deque
def solution(rows, columns, queries):
    # 행렬 초기화
    answer = []
    board = []
    for y in range(rows):
        b = []
        for x in range(columns):
            b.append((y*columns) + (x+1))
        board.append(b)

    # 회전
    for y1, x1, y2, x2 in queries:
        y1, x1, y2, x2 = y1-1, x1-1, y2-1, x2-1
        tmp = deque()

        for y in range(y1, y2):
            tmp.append(board[y][x1])
        for x in range(x1,x2):
            tmp.append(board[y2][x])
        for y in range(y2, y1, -1):
            tmp.append(board[y][x2])
        for x in range(x2, x1, -1):
            tmp.append(board[y1][x])
        
        tmp.append(tmp.popleft())
        answer.append(min(tmp))
        
        for y in range(y1, y2):
            board[y][x1] = tmp.popleft()
        for x in range(x1,x2):
            board[y2][x] = tmp.popleft()
        for y in range(y2, y1,-1):
            board[y][x2] = tmp.popleft()
        for x in range(x2, x1, -1):
            board[y1][x] = tmp.popleft()
    return answer



print(solution(6,6,[[2,2,5,4],[3,3,6,6],[5,1,6,3]])) # [8, 10, 25]
print(solution(3,3,[[1,1,2,2],[1,2,2,3],[2,1,3,2],[2,2,3,3]])) # [1, 1, 5, 3]
print(solution(100,97,[[1,1,100,97]])) #[1]